### ISSUE TITLE
Minor PEP8 (and other) issues

### ISSUE THREAD

**Issue by maciejors** [krezelj/academia#151]
**tldr: Some of the code looks messy. We could address most of the issues with one click in PyCharm once all the important work is done**

As a PyCharm user with a perfectionist's soul I am often bothered by some of the files that show 20 or so warnings with squiggly lines all over them. At the start I would point these issues out in reviews but I stopped doing that because I found it counterproductive - these are minor issues which don't break anything and are super quick to fix but delay merging and can potentially overshadow more serious issues with the code. 

I believe it's important to keep the code clean so that it looks professional. On the other hand however, as what we're working on is still pre-release stuff I don't think it's a big deal if the code looks a bit untidy now because we can resolve these issues with a single commit each just before publishing the package and the code. 

I propose that at the very end, each of us opens up this project in PyCharm, goes through the files that they've worked on (in order not to mess up git blame) and addresses all the minor issues like "too many empty lines" or "Instance attribute state defined outside __init__" or anything else that is easy-to-fix and at the same time makes the code look cleaner and therefore more professional. These problems are easy to spot in Pycharm as you can list and go through them one-by-one by clicking the warning icon on the right hand side at the top of the editor. Additionally most of PEP8-related issues can be fixed by pressing ctrl+alt+shift+L and then hitting the "run" button (reformat file).

Keep in mind though that no tool is ideal including PyCharm so some warnings can be false flags (from my experience 99% false positives are related to object types).
